tests/unittest.py

```diff
@@ -223,21 +215,92 @@ def assertIsTopological(self, tx_sequence: Iterator[BaseTransaction], message: O
+        self.assertNotEqual(s1, s2)
 
-        s1 = set(manager1.tx_storage.get_tx_tips())
-        s2 = set(manager2.tx_storage.get_tx_tips())
+    def assertTipsEqualSyncV1(self, manager1, manager2):
+        # tx tips
+        tips1 = {tx.hash for tx in manager1.tx_storage.iter_mempool_tips_from_tx_tips()}
+        tips2 = {tx.hash for tx in manager2.tx_storage.iter_mempool_tips_from_tx_tips()}
+        self.log.debug('tx tips1', len=len(tips1), list=shorten_hash(tips1))
+        self.log.debug('tx tips2', len=len(tips2), list=shorten_hash(tips2))
+        self.assertEqual(tips1, tips2)
+
+        # best block
+        s1 = set(manager1.tx_storage.get_best_block_tips())
+        s2 = set(manager2.tx_storage.get_best_block_tips())
         self.assertEqual(s1, s2)
 
-    def assertTipsNotEqual(self, manager1, manager2):
+        # best block (from height index)
+        b1 = manager1.tx_storage.indexes.height.get_tip()
+        b2 = manager2.tx_storage.indexes.height.get_tip()
+        self.assertEqual(b1, b2)
+
+        # all tips must be equal (this check should be removed together with the index)
         s1 = set(manager1.tx_storage.get_all_tips())
         s2 = set(manager2.tx_storage.get_all_tips())
-        self.assertNotEqual(s1, s2)
+        self.assertEqual(s1, s2)
+
+    def assertTipsEqualSyncV2(self, manager1, manager2, *, strict_sync_v2_indexes=True):
+        # tx tips
+        if strict_sync_v2_indexes:
+            tips1 = manager1.tx_storage.indexes.mempool_tips.get()
+            tips2 = manager2.tx_storage.indexes.mempool_tips.get()
+        else:
+            tips1 = {tx.hash for tx in manager1.tx_storage.iter_mempool_tips_from_best_index()}
+            tips2 = {tx.hash for tx in manager2.tx_storage.iter_mempool_tips_from_best_index()}
+        self.log.debug('tx tips1', len=len(tips1), list=shorten_hash(tips1))
+        self.log.debug('tx tips2', len=len(tips2), list=shorten_hash(tips2))
+        # self.assertEqual(tips1, tips2)
+        # XXX: this is still not correct, tips may diverge since voided transactions aren't propagated
+        # XXX: temporarily disabled because tips may not match since sync-v2-mempool isn't merged yet
+        if len(tips1) < len(tips2):
+            self.assertEqual(tips1, tips1 & tips2)
+        else:
+            self.assertEqual(tips2, tips1 & tips2)
+
+        # best block
+        s1 = set(manager1.tx_storage.get_best_block_tips())
+        s2 = set(manager2.tx_storage.get_best_block_tips())
+        self.log.debug('block tips1', len=len(s1), list=shorten_hash(s1))
+        self.log.debug('block tips2', len=len(s2), list=shorten_hash(s2))
+        self.assertEqual(s1, s2)
+
+        # best block (from height index)
+        b1 = manager1.tx_storage.indexes.height.get_tip()
+        b2 = manager2.tx_storage.indexes.height.get_tip()
+        self.assertEqual(b1, b2)
 
     def assertConsensusEqual(self, manager1, manager2):
+        _, enable_sync_v2 = self._syncVersionFlags()
```

Comment:
Why don't we check which version is really being used by manager1 and manager2? We should also assert that both of them are returning the same version.

Reply:
I'll add a check to see if the sync version from the `_syncVersionFlags` is consistent with the sync version enabled on each of the given manager. Which is not exactly the same as checking if they are "returning the same version", because they are independent, both versions can be enabled (in that case the sync_v2 check is used, because that is what's expected that both nodes will use to sync).

Reply:
Outdated.